agglayer/grpc/agglayer_grpc_client.go

```diff
@@ -404,3 +460,21 @@ func certificateStatusFromProto(status v1nodetypes.CertificateStatus) types.Cert
 		return types.Pending
 	}
 }
+
+// handleGrpcError checks if the error is a retryable gRPC error
+// and returns a formatted error message.
+func handleGrpcError(err error) error {
+	if err != nil {
+		if !isRetryableGRPCError(err) {
+			return fmt.Errorf("%w: %w", aggkitcommon.ErrNonRetryable, err)
+		}
+		return fmt.Errorf("transient error: %w", err)
+	}
+	return nil
+}
+
+// isRetryableGRPCError checks if the error is a retryable gRPC error
+func isRetryableGRPCError(err error) bool {
```

Comment:
I would move this to the `grpc_client.go` alongside the `handleGrpcError`. It feels more natural to be there.